R/pkg/R/DataFrame.R

```diff
@@ -3939,7 +3929,15 @@ setMethod("hint",
           signature(x = "SparkDataFrame", name = "character"),
           function(x, name, ...) {
             parameters <- list(...)
-            stopifnot(all(sapply(parameters, isTypeAllowedForSqlHint)))
+            stopifnot(all(sapply(parameters, function(x) {
```

Comment:
if recall, let's not have a inside scope with the same variable name `x` in the outer scope?

Reply:
My bad. Will change.